spec/components/jobseekers/job_applications/detail_component_spec.rb

```diff
@@ -0,0 +1,41 @@
+require "rails_helper"
+
+RSpec.describe Jobseekers::JobApplications::DetailComponent, type: :component do
+  let(:detail) do
+    create(:job_application_detail, details_type: "talents", data: { "name" => "John", "speciality" => "Strength" })
+  end
+  let(:info_to_display) do
+    [
+      { attribute: "name", title: "Full name" },
+      { attribute: "speciality", title: "Superpower(s)" },
+    ]
+  end
+
+  let!(:inline_component) do
+    render_inline(described_class.new(detail: detail, detail_counter: 1, info_to_display: info_to_display))
+  end
+
+  it "renders the name" do
+    expect(inline_component.css(".govuk-summary-list__row").first.to_html)
+      .to eq("<div class=\"govuk-summary-list__row\">\n      <dt class=\"govuk-summary-list__key\">\n        Full name\n      </dt>\n\n      <dd class=\"govuk-summary-list__value\">\n        John\n      </dd>\n\n</div>")
+  end
+
+  it "renders the speciality" do
+    expect(inline_component.css(".govuk-summary-list__row").last.to_html)
+      .to eq("<div class=\"govuk-summary-list__row\">\n      <dt class=\"govuk-summary-list__key\">\n        Superpower(s)\n      </dt>\n\n      <dd class=\"govuk-summary-list__value\">\n        Strength\n      </dd>\n\n</div>")
```

Comment:
Might be a bit too much to test for the whole HTML output, because if you change it a bit, then you have to regenerate it by copying and pasting it. I would be happy to just test the content, not the style, but I would leave the final decision to you